pkg/cosign/verify.go

```diff
@@ -71,8 +71,10 @@ type CheckOpts struct {
 
 	// Annotations optionally specifies image signature annotations to verify.
 	Annotations map[string]interface{}
+	// CertExtensions optionally specifies image signature cert extensions to verify.
+	CertExtensions map[string]string
```

Comment:
In the same vein, moving away from a map, I think we should have a field per extension - CertGitHubSha, CertGitHubRef, etc - If this is used a library function, it's not as clean of an interface to take a map, because there's no way to know what is a valid string.